src/panels/Menu/TopMenuBar/TaskBarMenuChoices.tsx

```diff
@@ -0,0 +1,66 @@
+import { Dispatch, SetStateAction } from 'react';
+import { CheckboxIndicator, Menu } from '@mantine/core';
+
+import { ChevronRightIcon, TaskBarIcon } from '@/icons/icons';
+import { IconSize } from '@/types/enums';
+import { menuItemsData } from '@/windowmanagement/data/MenuItems';
+
+import { MenuWrapper } from './MenuWrapper';
+
+interface TaskBarChoicesProps {
+  visibleMenuItems: string[];
+  setVisibleMenuItems: Dispatch<SetStateAction<string[]>>;
+}
+
+export function TaskBarMenuChoices({
+  visibleMenuItems,
+  setVisibleMenuItems
+}: TaskBarChoicesProps) {
+  // Mapping menu item ids to boolean flag to determine their checked status
+  const checkedMenuItems: { [key: string]: boolean } = {};
+  visibleMenuItems.forEach((menuItemID) => {
+    checkedMenuItems[menuItemID] = true;
+  });
+
+  function toggleMenuItem(id: string): void {
+    setVisibleMenuItems((prevstate) => {
+      const index = prevstate.indexOf(id);
+      const isChecked = index !== -1;
+
+      if (isChecked) {
+        prevstate.splice(index, 1);
+        return [...prevstate];
+      } else {
+        return [...prevstate, id];
+      }
+    });
```

Comment:
Consider using mantines hook for state objects, then you dont have to do a deep copy https://mantine.dev/hooks/use-set-state/. Then you could use an object instead of an array with the `visible[id] = true` for example. That would also make the check above easier since you have a O(1) map rather than the array